ecommerce_extensions/tenant/tests/test_extra_scripts.py

```diff
@@ -0,0 +1,109 @@
+"""
+Tests for the extra scripts.
+"""
+from __future__ import absolute_import, unicode_literals
+
+from django.test import TestCase
+from path import Path
+from testfixtures import LogCapture
+
+from ecommerce_extensions.tenant.extra_scripts import process_scripts, validate_script_attributes
+
+
+class TestsProcessExtraScripts(TestCase):
+    """ Tests for extra scripts"""
+
+    def test_returned_process_scripts(self):
+        """
+        Test process_scripts function returns a dictionary with all the valid scripts
+        in the correct order.
+        """
+        path = Path("/test")
+        tenantoptions = {
+            "scripts": {
+                ".*/test": [
+                    {
+                        "src": "https://www.test-external-script.com/js/myScript1.js",
+                        "type": "external",
+                        "media_type": "module"
+                    },
+                    {
+                        "content": "alert('This alert box was called with the onload event');",
+                        "type": "inline"
+                    }
+                ]
+            }
+        }
+        test_scripts = [
+            {
+                "src": "https://www.test-external-script.com/js/myScript1.js",
+                "type": "external",
+                "media_type": "module"
+            },
+            {
+                "content": "alert('This alert box was called with the onload event');",
+                "type": "inline",
+                "media_type": "text/javascript"
+            },
+        ]
+
+        scripts = process_scripts(path, tenantoptions)
+
+        self.assertEqual(test_scripts, scripts)
+
+    def test_returned_(self):
+        """
+        Test process_scripts function returns only the scripts that have a valid configuration.
+        """
+        path = Path("/test")
+        tenantoptions = {
+            "scripts": {
+                ".*/test": [
+                    {
+                        "content": "alert('This alert box was called with the onload event');",
+                        "type": "src"
+                    }
+                ]
+            }
+        }
+
+        scripts = process_scripts(path, tenantoptions)
+
+        self.assertEqual(None, scripts)
```

Comment:
@andrey-canon Remember that this was already implemented in eox-theming, I simply adapted the logic for ecommerce-extension. I thought that if I did it like that, I would have less chance of errors, instead of reinventing the wheel. Since at some point you must have checked that code.

In any case, if you want, I can make a different proposal.

1. I think it shouldn't be possible. If the `type` is `internal`, the `media_type` could only be `text/javascript`, otherwise it could only be `module`. In this sense, perhaps `type` is not even necessary, because indicating the `media_type` would know what to expect in `content` or `src`. Even easier could be to simply indicate `content` or `src` and the rest to be interpreted by the code.
2. I assume that `type` does not have a default value, because it is the attribute that indicates how the content will be interpreted.
3. I do not know what you are talking about. What permission file to change?

Reply:
to sum up, this is a copy from other code and you should specify that in the pr description the next time that you do that. I'm not happy with this code since imo this to complex and should be refactored, however that is not the purpose of this pr, so my unique suggestion is to restore the file permissions

![image](https://user-images.githubusercontent.com/36200299/117194440-b99bb600-ada9-11eb-8fc5-3dd0ef84580b.png)
the numbers at the end mean that you changed the file permissions

Reply:
Understood for the next one. Ready what you request.

Reply:
I do think that `type` is unnecessary and deleting it would make the code much more simple. why not implement the code here with this fix at once?